siriuspy/siriuspy/pwrsupply/prucontroller.py

```diff
@@ -1042,10 +1043,15 @@ def _bsmp_update_variables(self, device_ids, group_id):
                     copy_var_vals[id][self._params.ConstBSMP.V_I_LOAD2] += \
                         0.00001*_random.uniform(-1.0, +1.0)
 
+            elif ack[id] == _Response.invalid_id:
+                self._connected[id] = False
+                create_groups = True
             else:
                 self._connected[id] = False
         # processing time up to this point: 19.4 ms @ BBB1
         # print('time3: ', _time.time() - t0)
+        if create_groups:
+            self._bsmp_init_groups()
```

Comment:
@gu1lhermelp , do we need to recreate variables groups for all BSMP devices? maybe *create_groups* should be turned into a list of BSMP IDs for which variables groups need to be recreated and then passed as argument for a generalized <code>self._bsmp_init_groups(list_ids=None)</code>